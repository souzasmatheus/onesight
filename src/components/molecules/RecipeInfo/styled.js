import styled from 'styled-components';
import Avatar from '@material-ui/core/Avatar';
import CardMedia from '@material-ui/core/CardMedia';
import Card from '@material-ui/core/Card';
import TableRow from '@material-ui/core/TableRow';

export const StyledRow = styled(TableRow)`
  cursor: pointer;
`;

export const StyledAvatar = styled(Avatar)`
  margin: 0 auto;
`;

export const StyledCardMedia = styled(CardMedia)`
  height: 150px;
`;

export const StyledCard = styled(Card)`
  max-width: 550px;
`;
