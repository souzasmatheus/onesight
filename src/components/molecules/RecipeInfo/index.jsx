import React, { useState } from 'react';
import PropTypes from 'prop-types';

import TableCell from '@material-ui/core/TableCell';
import IconButton from '@material-ui/core/IconButton';
import Dialog from '@material-ui/core/Dialog';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import VideoIcon from '@material-ui/icons/OndemandVideo';
import CloseIcon from '@material-ui/icons/Close';

import { StyledAvatar, StyledCardMedia, StyledCard, StyledRow } from './styled';

const RecipeInfo = ({ src, name, category, link, description }) => {
  const [showDetails, setShowDetails] = useState(false);

  return (
    <>
      <StyledRow>
        <TableCell
          data-testid="recipe-table-cell"
          onClick={() => setShowDetails(true)}
          align="center"
        >
          <StyledAvatar src={`${src}/preview`} />
        </TableCell>
        <TableCell onClick={() => setShowDetails(true)} align="center">
          {name}
        </TableCell>
        <TableCell onClick={() => setShowDetails(true)} align="center">
          {category}
        </TableCell>
        <TableCell align="center">
          <IconButton
            onClick={() => {
              window.open(link);
            }}
          >
            <VideoIcon />
          </IconButton>
        </TableCell>
      </StyledRow>
      <Dialog open={showDetails} onClose={() => setShowDetails(false)}>
        <StyledCard>
          <CardHeader
            title={name}
            subheader={category}
            action={
              <IconButton onClick={() => setShowDetails(false)}>
                <CloseIcon />
              </IconButton>
            }
          />
          <StyledCardMedia image={src} title={name} />
          <CardContent>
            <Typography>{description}</Typography>
          </CardContent>
        </StyledCard>
      </Dialog>
    </>
  );
};

RecipeInfo.propTypes = {
  name: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  src: PropTypes.string,
  link: PropTypes.string,
  description: PropTypes.string,
};

RecipeInfo.defaultProps = {
  src: '',
  link: '',
  description: '',
};

export default RecipeInfo;
