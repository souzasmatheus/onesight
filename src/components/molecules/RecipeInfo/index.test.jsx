import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { MockedStore } from '~/../test/mocks/providers';

import RecipeInfo from './index';

describe('<RecipeInfo />', () => {
  const props = {
    name: 'Arrabiata',
    category: 'Italian',
    description: 'Una ricetta squisita',
  };

  it('should match snapshot test', () => {
    const tree = render(
      <MockedStore>
        <RecipeInfo {...props} />
      </MockedStore>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render information correctly', () => {
    const { getByText } = render(
      <MockedStore>
        <RecipeInfo {...props} />
      </MockedStore>,
    );

    expect(getByText(props.name)).toBeDefined();
    expect(getByText(props.category)).toBeDefined();
  });

  it('should open Dialog on click', () => {
    const { getByTestId, getByText } = render(
      <MockedStore>
        <RecipeInfo {...props} />
      </MockedStore>,
    );

    fireEvent.click(getByTestId('recipe-table-cell'));

    expect(getByText(props.description)).toBeDefined();
  });
});
