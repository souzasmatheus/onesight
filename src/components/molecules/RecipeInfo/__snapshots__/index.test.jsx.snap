// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeInfo /> should match snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="MuiTableRow-root sc-bdfBwQ iVZKII"
      >
        <td
          class="MuiTableCell-root MuiTableCell-alignCenter"
          data-testid="recipe-table-cell"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle sc-gsTCUz hZKgGY"
          >
            <img
              class="MuiAvatar-img"
              src="/preview"
            />
          </div>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignCenter"
        >
          Arrabiata
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignCenter"
        >
          Italian
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignCenter"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 3H3c-1.11 0-2 .89-2 2v12c0 1.1.89 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.11-.9-2-2-2zm0 14H3V5h18v12zm-5-6l-7 4V7z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="MuiTableRow-root sc-bdfBwQ iVZKII"
    >
      <td
        class="MuiTableCell-root MuiTableCell-alignCenter"
        data-testid="recipe-table-cell"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle sc-gsTCUz hZKgGY"
        >
          <img
            class="MuiAvatar-img"
            src="/preview"
          />
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-alignCenter"
      >
        Arrabiata
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-alignCenter"
      >
        Italian
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-alignCenter"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 3H3c-1.11 0-2 .89-2 2v12c0 1.1.89 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.11-.9-2-2-2zm0 14H3V5h18v12zm-5-6l-7 4V7z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
